### PR TITLE
add dbt_valid_to_current 

### LINKED ISSUE #6275: [Core] Allow custom date for `dbt_valid_to` in snapshots
Add support for the `dbt_valid_to_current` specification for snapshot records, along with updates to tests and schemas. The most important changes include updates to the snapshot configuration, schema files, and various test files.

Originating from this issue: https://github.com/dbt-labs/dbt-core/issues/10187
Pull request: https://github.com/dbt-labs/dbt-core/pull/10780

### PR DESCRIPTION
**Pull request by mirnawong1**
adding new config for `dbt_valid_to_current`, which allows users to set a future date for `dbt_valid_to` using the dbt_valid_to_current config in yaml/config/project.

Resolves #6275

Outstanding questions:
✅ - Can date be hardcoded using this syntax? "to_date('2024-05-10')" or can they use "to_date('2024, 05, 10')"  ?
✅ - is the syntax just a string (e.g. `dbt_valid_to_current: '9999, 12, 31'
`) or must they use `to_date`?
✅ - can user use `var` to return a SQL statement '{{ var('my_future_date') }}'
✅ - can they use a macro that returns a SQL statement '{{ dbt.date(9999, 12, 31) }}'?
✅ - how does this new config work with deferral/state:modified? will we warn users that the config has been updated and they need to manually update their snapshot?

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-add-custom-date-dbt-labs.vercel.app/docs/build/snapshots
- https://docs-getdbt-com-git-add-custom-date-dbt-labs.vercel.app/docs/dbt-versions/release-notes
- https://docs-getdbt-com-git-add-custom-date-dbt-labs.vercel.app/reference/resource-configs/dbt_valid_to_current

### CODE REVIEW

**Comment by dbeatty10** [comment]
@mirnawong1 grouping your outstanding questions into a handful of categories:

### Hardcoded SQL expressions

Question:
- Can date be hardcoded using this syntax? "to_date('2024-05-10')" or can they use "to_date('2024, 05, 10')" ?

👍 Hardcoded expressions that work within the data platform is the _only_ option that is available currently. The precise syntax will vary per warehouse, so `to_date('2024-05-10')` may work for some, but others may require something like `date(2024, 5, 10)`.

### Jinja within `dbt_valid_to_current`

Questions:
- is the syntax just a string (e.g. dbt_valid_to_current: '9999, 12, 31' ) or must they use to_date?
- can user use var to return a SQL statement '{{ var('my_future_date') }}'
- can they use a macro that returns a SQL statement '{{ dbt.date(9999, 12, 31) }}'?

👎 Jinja is not currently available for `dbt_valid_to_current`. So none of the above are possible, and the only option is to supply a full SQL expression that is specific to the users data warehouse.

### Deferral / `state:modified`

Question:
- how does this new config work with deferral/state:modified? will we warn users that the config has been updated and they need to manually update their snapshot?

👍 Based on hands-on experimentation, it **does** work as expected with deferral / `state:modified`. So if this config changes, than it will show up when using `--select state:modified`.

**Comment by dbeatty10** [comment]
@mirnawong1 I added the `blocked_by_dev` label until this is resolved:
- https://github.com/dbt-labs/dbt-core/issues/10921

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+This simplifies makes it easier to assign date, work in a join, and range-based filtering with an end date.
+
+## Default
+
+By default, `dbt_valid_to` is set to `NULL` for current (most recent) records in your snapshot table. This means that these records are still valid and have no defined end date.
+
+If you prefer to use a specific value instead of `NULL` for `dbt_valid_to` in current and future records, you can use the `dbt_valid_to_current` configuration option. For example, setting a date in the far future, `9999-12-31`.
+
+The value assigned to `dbt_valid_to_current` should be a string representing a valid date or timestamp, depending on your database's requirements.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
The value assigned to `dbt_valid_to_current` should be a string representing a valid date or timestamp, depending on your database's requirements. Use expressions that work within the data platform.
```

**Comment by mirnawong1** [comment]
> @mirnawong1 I added the `blocked_by_dev` label until this is resolved:
> 
> * [[Bug] When a custom `dbt_valid_to_current` is configured, it doesn't update dbt-core#10921](https://github.com/dbt-labs/dbt-core/issues/10921)

thank you @dbeatty10 ! it looks like this is resolved now right?

File: website/docs/docs/build/snapshots.md [hunk 2]
```diff
@@ -204,12 +209,14 @@ Snapshots can't be rebuilt. Because of this, it's a good idea to put snapshots i
...
 * **On subsequent runs:** dbt will check which records have changed or if any new records have been created:
-  - The `dbt_valid_to` column will be updated for any existing records that have changed
-  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null`
+  - The `dbt_valid_to` column will be updated for any existing records that have changed.
+  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null` or the value configured in `dbt_valid_to_current`.
 
-Note, these column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+#### Note 
+- These column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+- If you have set the `dbt_valid_to_current` configuration option, then instead of `NULL`, the `dbt_valid_to` field in future records will be set to your specified value (such as `9999-12-31`).
```
Comment (dbeatty10) [inline website/docs/docs/build/snapshots.md]: @mirna this content is showing when I set my docs version to v1.8.

Does it need to be in a Version block or have some other call-out about the relevant version?

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+---
+resource_types: [snapshots]
+description: "Snapshot dbt_valid_to_current custom date"
+datatype: "{<dictionary>}"
+default_value: {"dbt_valid_from": "dbt_valid_from", "dbt_valid_to": "dbt_valid_to", "dbt_scd_id": "dbt_scd_id", "dbt_updated_at": "dbt_updated_at"}
```
Comment (dbeatty10) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: The default value for `dbt_valid_to_current` is `NULL`.

I'm guessing this was a copy-paste carry-over from the docs for `snapshot_meta_column_names`.

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    +dbt_valid_to_current: "to_date('9999-12-31')"
+```
+
+</File>
+
+## Description
+
+Use the `dbt_valid_to_current` config to set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
+
+This simplifies makes it easier to assign date, work in a join, and range-based filtering with an end date.
```
Comment (dbeatty10) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: This sentence sounds funny, but I don't know how to re-work it. But I bet you'll have some ideas 🧠

**Review by dbeatty10 — APPROVED** [review]
I didn't test out the code examples, but looks good to me in general.

Left a couple comments on some small non-blocking items.

File: website/docs/reference/resource-configs/dbt_valid_to_current.md (hunk 3 above)
Comment (mirnawong1) [reply website/docs/reference/resource-configs/dbt_valid_to_current.md]: good shout!

File: website/docs/docs/build/snapshots.md (hunk 2 above)
Comment (mirnawong1) [reply website/docs/docs/build/snapshots.md]: nice catch, thanks @dbeatty10 ! have added a callout

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+</File>
+
+## Description
+
+Use the `dbt_valid_to_current` config to set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
+
+This approach makes it easier to assign a custom date date, work in a join, or perform range-based filtering that require an end date.
+
+## Usage
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
### Considerations
```

File: website/docs/docs/build/snapshots.md [hunk 6]
```diff
@@ -87,13 +81,14 @@ The following table outlines the configurations available for snapshots:
 | [check_cols](/reference/resource-configs/check_cols) | If using the `check` strategy, then the columns to check | Only if using the `check` strategy | ["status"] |
 | [updated_at](/reference/resource-configs/updated_at) | If using the `timestamp` strategy, the timestamp column to compare | Only if using the `timestamp` strategy | updated_at |
 | [invalidate_hard_deletes](/reference/resource-configs/invalidate_hard_deletes) | Find hard deleted records in source and set `dbt_valid_to` to current time if the record no longer exists | No | True |
+| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table. | No | string |
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]: Instead of "custom future date", I would actually describe this as something like "set a custom indicator for what `dbt_valid_to` should be for current records" (but better phrasing lol) and we can call out that the default is `NULL`

File: website/docs/docs/build/snapshots.md [hunk 7]
```diff
@@ -172,6 +168,15 @@ This strategy handles column additions and deletions better than the `check` str
 
 </Expandable>
 
+
+<Expandable alt_header="Use dbt_valid_to_current for easier date range queries">
+
+By default, `dbt_valid_to` is `NULL` for current records. However, if you set the [`dbt_valid_to_current` configuration](/reference/resource-configs/dbt_valid_to_current) (available in Versionless and 1.9 and higher), `dbt_valid_to` will be set to your specified value (such as `9999-12-31`) for current records.
+
+This simplifies your SQL queries by avoiding `NULL` checks and allowing for straightforward date range filtering.
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]: What does "simplifies your SQL queries by avoiding `NULL` checks" mean?

File: website/docs/docs/build/snapshots.md [hunk 8]
```diff
@@ -204,12 +209,14 @@ Snapshots can't be rebuilt. Because of this, it's a good idea to put snapshots i
...
 * **On subsequent runs:** dbt will check which records have changed or if any new records have been created:
-  - The `dbt_valid_to` column will be updated for any existing records that have changed
-  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null`
+  - The `dbt_valid_to` column will be updated for any existing records that have changed.
+  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null` or the value configured in `dbt_valid_to_current` (available in Versionless and 1.9 and higher).
 
-Note, these column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+#### Note 
+- These column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+- If you have set the `dbt_valid_to_current` configuration option, then instead of `NULL`, the `dbt_valid_to` field in future records will be set to your specified value (such as `9999-12-31`).
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]: I think we're missing some clarity that this is set to `NULL` (or whatever you have set `dbt_valid_to_current` to for **current** records)

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
+---
+resource_types: [snapshots]
+description: "Snapshot dbt_valid_to_current custom date"
+datatype: "{<dictionary>}"
+default_value: {NULL}
+id: "dbt_valid_to_current"
+---
+
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md:8]: I don't know where we want this, but we should probably have a [migration callout/warning](https://docs.getdbt.com/reference/resource-configs/snapshot_meta_column_names#:~:text=To%20avoid%20any,column%20name%20change.) like what we have for the other new configs cc: @dbeatty10

File: website/docs/docs/build/snapshots.md [hunk 10]
```diff
@@ -394,12 +401,14 @@ snapshots:
 
 Snapshot <Term id="table">tables</Term> will be created as a clone of your source dataset, plus some additional meta-fields*.
 
-Starting in 1.9 or with [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), these column names can be customized to your team or organizational conventions via the [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) config.
+Starting in 1.9 or with [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless):
+- These column names can be customized to your team or organizational conventions using the [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) config. 
+- Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]: again, i would rephrase this as suggested above

File: website/docs/docs/dbt-versions/release-notes.md [hunk 11]
```diff
@@ -19,6 +19,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 \* The official release date for this new format of release notes is May 15th, 2024. Historical release notes for prior dates may not reflect all available features released earlier this year or their tenancy availability.
 
 ## November 2024
+- **New**: Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/docs/dbt-versions/release-notes.md]: again, rephrase as suggested above

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+snapshots:
+  [<resource-path>](/reference/resource-configs/resource-path):
+    +dbt_valid_to_current: "to_date('9999-12-31')"
+```
+
+</File>
+
+## Description
+
+Use the `dbt_valid_to_current` config to set a custom future date for `dbt_valid_to` in new snapshot columns. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: again, rephrase as suggested above

thinking about this more, i **do** like that we're calling out that the main use case for this is "a future date"

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+This approach makes it easier to assign a custom date date, work in a join, or perform range-based filtering that require an end date.
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: "date date"

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+## Default
+
+By default, `dbt_valid_to` is set to `NULL` for current (most recent) records in your snapshot table. This means that these records are still valid and have no defined end date.
+
+If you prefer to use a specific value instead of `NULL` for `dbt_valid_to` in current and future records, you can use the `dbt_valid_to_current` configuration option. For example, setting a date in the far future, `9999-12-31`.
+
+The value assigned to `dbt_valid_to_current` should be a string representing a valid date or timestamp, depending on your database's requirements. Use expressions that work within the data platform.
+
+### Managing records
+- **For existing records** &mdash; To avoid any unintentional data modification, dbt will _not_ automatically adjust the current value in the existing `dbt_valid_to` column. Existing current records will still have `dbt_valid_to` set to `NULL`.
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md:82]: Ah here's the migration callout - I think we just want to make sure this warning exists elsewhere as well!

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+## Default
+
+By default, `dbt_valid_to` is set to `NULL` for current (most recent) records in your snapshot table. This means that these records are still valid and have no defined end date.
+
+If you prefer to use a specific value instead of `NULL` for `dbt_valid_to` in current and future records, you can use the `dbt_valid_to_current` configuration option. For example, setting a date in the far future, `9999-12-31`.
+
+The value assigned to `dbt_valid_to_current` should be a string representing a valid date or timestamp, depending on your database's requirements. Use expressions that work within the data platform.
+
+### Managing records
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: Is there a more specific descriptor here for the heading?

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+- **For existing records** &mdash; To avoid any unintentional data modification, dbt will _not_ automatically adjust the current value in the existing `dbt_valid_to` column. Existing current records will still have `dbt_valid_to` set to `NULL`.
+
+- **For new records** &mdash;  Any new records inserted after applying the `dbt_valid_to_current` configuration will have `dbt_valid_to` set to the specified value (for example, '9999-12-31'), instead of `NULL`.
+
+This means your snapshot table will have current records with `dbt_valid_to` values of both `NULL` (from existing data) and the new specified value (from new data). If you'd rather have consistent `dbt_valid_to` values for current records, you can either manually update existing records in your snapshot table where `dbt_valid_to` is `NULL` to match your `dbt_valid_to_current` value or rebuild your snapshot table.
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: "rebuild your snapshot table" would be very risky (lose all of your historical data!)

**Review by graciegoheen — DISMISSED** [review]
Few comments but this is looking really good, thank you!

File: website/docs/reference/resource-configs/dbt_valid_to_current.md (hunk 13 above)
Comment (mirnawong1) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
This approach makes it easier to assign a custom date, work in a join, or perform range-based filtering that requires an end date.
```

File: website/docs/docs/build/snapshots.md (hunk 6 above)
Comment (mirnawong1) [reply website/docs/docs/build/snapshots.md]: will use a general 'custom indicator' description and use custom future date as example

Comment (mirnawong1) [inline website/docs/docs/build/snapshots.md]:
```suggestion
| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records. By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.| No | string |
```

File: website/docs/docs/build/snapshots.md [hunk 17]
```diff
@@ -87,13 +81,14 @@ The following table outlines the configurations available for snapshots:
 | [check_cols](/reference/resource-configs/check_cols) | If using the `check` strategy, then the columns to check | Only if using the `check` strategy | ["status"] |
 | [updated_at](/reference/resource-configs/updated_at) | If using the `timestamp` strategy, the timestamp column to compare | Only if using the `timestamp` strategy | updated_at |
 | [invalidate_hard_deletes](/reference/resource-configs/invalidate_hard_deletes) | Find hard deleted records in source and set `dbt_valid_to` to current time if the record no longer exists | No | True |
+| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records. By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.| No | string |
```
Comment (mirnawong1) [inline website/docs/docs/build/snapshots.md]: @graciegoheen , it looks like [gerda says](https://dbt-labs.slack.com/archives/C05FWBP9X1U/p1729105620164219?thread_ts=1729072739.374999&cid=C05FWBP9X1U) the config applies to new snapshot columns, should i add that here?

```suggestion
| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records. By default, this value is `NULL`. When configured, dbt will use the specified value for `dbt_valid_to` in new snapshot columns, but it will not automatically adjust the value in existing columns. Existing columns that use `NULL` can be updated manually if needed.  | No | string |
```

File: website/docs/docs/build/snapshots.md (hunk 7 above)
Comment (mirnawong1) [reply website/docs/docs/build/snapshots.md]: i meant users not needing to include checks or add'l logic to check for null records. 

i can remove if it's confusing or clarify further!

File: website/docs/docs/build/snapshots.md (hunk 17 above)
Comment (mirnawong1) [inline website/docs/docs/build/snapshots.md]:
```suggestion
| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.| No | string |
```

File: website/docs/docs/build/snapshots.md [hunk 18]
```diff
@@ -87,13 +81,14 @@ The following table outlines the configurations available for snapshots:
 | [check_cols](/reference/resource-configs/check_cols) | If using the `check` strategy, then the columns to check | Only if using the `check` strategy | ["status"] |
 | [updated_at](/reference/resource-configs/updated_at) | If using the `timestamp` strategy, the timestamp column to compare | Only if using the `timestamp` strategy | updated_at |
 | [invalidate_hard_deletes](/reference/resource-configs/invalidate_hard_deletes) | Find hard deleted records in source and set `dbt_valid_to` to current time if the record no longer exists | No | True |
+| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.| No | string |
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]:
```suggestion
| [dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current) | Set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table.| No | string |
```

File: website/docs/docs/build/snapshots.md (hunk 7 above)
Comment (graciegoheen) [reply website/docs/docs/build/snapshots.md]: I think remove the null checks piece, and just say "This allows for straightforward date range filtering."

File: website/docs/docs/build/snapshots.md [hunk 19]
```diff
@@ -204,13 +209,15 @@ Snapshots can't be rebuilt. Because of this, it's a good idea to put snapshots i
...
-  - The `dbt_valid_to` column will be updated for any existing records that have changed
-  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null`
-
-Note, these column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+  - The `dbt_valid_to` column will be updated for any existing records that have changed.
+  - The updated record and any new records will be inserted into the snapshot table. These records will now have `dbt_valid_to = null` or the value configured in `dbt_valid_to_current` (available in Versionless and 1.9 and higher).
 
+#### Note 
+- These column names can be customized to your team or organizational conventions using the [snapshot_meta_column_names](#snapshot-meta-fields) config.
+- Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date such as `9999-12-31`). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]:
```suggestion
- Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date such as `9999-12-31`). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table.
```

File: website/docs/docs/build/snapshots.md [hunk 20]
```diff
@@ -394,12 +401,14 @@ snapshots:
 
 Snapshot <Term id="table">tables</Term> will be created as a clone of your source dataset, plus some additional meta-fields*.
 
-Starting in 1.9 or with [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), these column names can be customized to your team or organizational conventions via the [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) config.
+Starting in 1.9 or with [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless):
+- These column names can be customized to your team or organizational conventions using the [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) config. 
+- Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date such as `9999-12-31`). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/docs/build/snapshots.md]:
```suggestion
- Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date such as `9999-12-31`). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table.
```

File: website/docs/docs/dbt-versions/release-notes.md [hunk 21]
```diff
@@ -19,6 +19,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 \* The official release date for this new format of release notes is May 15th, 2024. Historical release notes for prior dates may not reflect all available features released earlier this year or their tenancy availability.
 
 ## November 2024
+- **New**: Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` in the snapshot table. This feature is available in dbt Cloud Versionless and dbt Core v1.9.
```
Comment (graciegoheen) [inline website/docs/docs/dbt-versions/release-notes.md]:
```suggestion
- **New**: Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table. This feature is available in dbt Cloud Versionless and dbt Core v1.9.
```

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 22]
```diff
@@ -0,0 +1,116 @@
+---
+resource_types: [snapshots]
+description: "Snapshot dbt_valid_to_current custom date"
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]: should we update this to custom current indicator?

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+snapshots:
+  [<resource-path>](/reference/resource-configs/resource-path):
+    +dbt_valid_to_current: "to_date('9999-12-31')"
+```
+
+</File>
+
+## Description
+
+Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` in the snapshot table.
```
Comment (graciegoheen) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When set, dbt will use this specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table.
```

**Review by graciegoheen — COMMENTED** [review]
a few nits!

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+
+## Impact on snapshot records
+
+When you set `dbt_valid_to_current`, it affects how dbt manages the `dbt_valid_to` column in your snapshot table:
+
+- **For existing records** &mdash; To avoid any unintentional data modification, dbt will _not_ automatically adjust the current value in the existing `dbt_valid_to` column. Existing current records will still have `dbt_valid_to` set to `NULL`.
+
+- **For new records** &mdash;  Any new records inserted after applying the `dbt_valid_to_current` configuration will have `dbt_valid_to` set to the specified value (for example, '9999-12-31'), instead of `NULL`.
+
+This means your snapshot table will have current records with `dbt_valid_to` values of both `NULL` (from existing data) and the new specified value (from new data). If you'd rather have consistent `dbt_valid_to` values for current records, you can manually update existing records in your snapshot table where `dbt_valid_to` is `NULL` to match your `dbt_valid_to_current` value.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
This means your snapshot table will have current records with `dbt_valid_to` values of both `NULL` (from existing data) and the new specified value (from new data). If you'd rather have consistent `dbt_valid_to` values for current records, you can manually update existing records in your snapshot table (where `dbt_valid_to` is `NULL`) to match your `dbt_valid_to_current` value.
```

File: website/docs/docs/dbt-versions/release-notes.md [hunk 25]
```diff
@@ -19,6 +19,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 \* The official release date for this new format of release notes is May 15th, 2024. Historical release notes for prior dates may not reflect all available features released earlier this year or their tenancy availability.
 
 ## November 2024
+- **New**: Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table. This feature is available in dbt Cloud Versionless and dbt Core v1.9.
```
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/release-notes.md]:
```suggestion
- **New**: Use the [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current) config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records (like a future date). By default, this value is `NULL`. When configured, dbt will use the specified value instead of `NULL` for `dbt_valid_to` for current records in the snapshot table. This feature is available in dbt Cloud Versionless and dbt Core v1.9 and later.
```

File: website/docs/reference/resource-configs/dbt_valid_to_current.md [hunk 26]
```diff
@@ -0,0 +1,116 @@
+---
+resource_types: [snapshots]
+description: "Use the `dbt_valid_to_current` config to set a custom indicator for the value of `dbt_valid_to` in current snapshot records"
+datatype: "{<dictionary>}"
+default_value: {NULL}
+id: "dbt_valid_to_current"
+---
+
+Available in 1.9 or with [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless) dbt Cloud.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/dbt_valid_to_current.md]:
```suggestion
Available from dbt v1.9 or with [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless) dbt Cloud.
```
